src/lib/viewers/image/ImageViewer.js

```diff
@@ -165,6 +188,36 @@ class ImageViewer extends ImageBaseViewer {
         };
     }
 
+    /**
+     * Gets the viewport dimensions.
+     *
+     * @return {Object} the width & height of the viewport
+     */
+
+    getViewportDimensions() {
+        return {
+            width: this.wrapperEl.clientWidth - 2 * IMAGE_PADDING,
+            height: this.wrapperEl.clientHeight - 2 * IMAGE_PADDING,
+        };
+    }
+
+    /**
+     * Sets mode to be AnnotationMode.NONE if the zoomed image overflows the viewport.
+     *
+     * @return {void}
+     */
+    handleZoomEvent({ newScale }) {
+        const width = newScale[0];
```

Comment:
Could these be restructured in one line?

Reply:
Yep will do.